internal/service/alarms/serve.go

```diff
@@ -201,3 +193,35 @@ func Serve(config *AlarmsServerConfig) error {
 
 	return nil
 }
+
+// UpdateAlarmDictionaryAndAlarmsDefinitionData reach out to cluster server and sync DB with alarm dict and def
+func UpdateAlarmDictionaryAndAlarmsDefinitionData(ctx context.Context, hubClient client.Client, a *internal.AlarmsServer) error {
```

Comment:
I'm always in favor of refactoring code to improve stuff. But this is too large (the whole dictionary refactoring) to be part of a single commit. To make review easier, please consider having a separate commit for this. Allain has been merging stuff with multiple commits (as far as I know, there is a label to avoid the squash). Or create a subsequent PR for this.

Or if you want to leave it like this, let's try to converge on the main purpose of this pr first. Then, I will switch context and review this refactoring.

Reply:
yeah let's leave this in one commit since i have testings everything together. review it in one shot thanks! should be very minimal changes...simply extracting out the functions and using errgroup to capture any error from the routine